apps/gameserver/src/Player.cpp

```diff
@@ -880,6 +1002,14 @@ void Player::Die()
     WriteInSight(bango::network::packet(S2C_ACTION, "db", GetID(), AT_DIE));
 }
 
+void Player::ResetStates()
+{
+    Character::ResetStates();
+    m_inventory = Inventory();
+    LeaveParty();
+    m_skills.clear();
```

Comment:
I think that skill manager should be done similarly to Inventory, instead of inheritance use member `m_skill_manager` field. This way we can hide m_skills map from the `Player` class.

```suggestion
    m_skills = SkillManager();
```